bin/prettier.js

```diff
@@ -361,30 +360,51 @@ function writeOutput(result) {
 }
 
 function eachFilename(patterns, callback) {
-  patterns.forEach(pattern => {
-    if (!glob.hasMagic(pattern)) {
-      if (shouldIgnorePattern(pattern)) {
-        return;
-      }
-      callback(pattern);
-      return;
-    }
+  if (ignoreNodeModules) {
+    patterns = patterns.concat(
+      ignoreNodeModulesGlobs.map(pattern => "!" + pattern)
+    );
+  }
 
-    glob(pattern, globOptions, (err, filenames) => {
-      if (err) {
-        console.error("Unable to expand glob pattern: " + pattern + "\n" + err);
-        // Don't exit the process if one pattern failed
-        process.exitCode = 2;
-        return;
+  return globby(patterns, globOptions)
+    .then(filePaths => {
+      if (filePaths.length === 0) {
+        const message =
+          (files => {
+            // Seperate files into last (last file) and initial - all the others
+            const initial = files.slice(0);
+            const last = initial.pop();
+
+            // Join into a comma seperated string of file names
+            const ending = files.length > 1
+              ? "and " + last + " do"
+              : last + " does";
+            return (initial.join(", ") + " " + ending).trim();
+          })(patterns) + " not match any files";
+
+        const err = new Error(message);
+        err.code = 2;
+        throw err;
       }
 
-      filenames.forEach(filename => {
-        callback(filename);
+      filePaths.forEach(filePath => {
+        return callback(filePath);
       });
-    });
-  });
-}
+    })
+    .catch(err => {
+      const isDoNotMatchAnyFilesError = /do not match any files/.test(
```

Comment:
This is a very fragile check. In fact, it is already incorrect, isn't it? The message can be `foo does not match any files` if there's only one pattern. Can't we just `console.error()` and set `process.exitCode` instead of `throw`ing this error above? If not, we could tag the error object with something to look for here.